src/components/vanilla/charts/TableChart/index.tsx

```diff
@@ -80,6 +81,12 @@ export default (props: Props) => {
     [maxRowsFit, props.results],
   );
 
+  useEffect(() => {
+    if (props.results?.data?.length) {
+      setHasNextPage(props.limit ? props.results.data.length >= props.limit : false);
```

Comment:
The logic ‎`props.results.data.length >= props.limit` will be ‎`true` even when the number of results exactly matches the limit. This may cause the “Next” button to be enabled even if there are no more results to fetch. I think that `=` should be removed then we are good to go